data/v2/base/controller.php

```diff
@@ -0,0 +1,401 @@
+<?php
+namespace Elementor\Data\V2\Base;
+
+use Elementor\Data\V2\Manager;
+use WP_REST_Controller;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * TODO: Utilize 'WP_REST_Controller' as much as possible.
+ */
+abstract class Controller extends WP_REST_Controller {
+	/**
+	 * Loaded endpoint(s).
+	 *
+	 * @var \Elementor\Data\V2\Base\Endpoint[]
+	 */
+	public $endpoints = [];
+
+	/**
+	 * Index endpoint.
+	 *
+	 * @var \Elementor\Data\V2\Base\Endpoint\Index
+	 */
+	public $index_endpoint = null;
+
+	/**
+	 * Loaded processor(s).
+	 *
+	 * @var \Elementor\Data\V2\Base\Processor[][]
+	 */
+	public $processors = [];
+
+	/**
+	 * @var \Elementor\Data\V2\Base\Controller|null
+	 */
+	private $parent = null;
+
+	/**
+	 * @var \Elementor\Data\V2\Base\Controller[]
+	 */
+	private $sub_controllers = [];
+
+	/**
+	 * Controller constructor.
+	 *
+	 * Register endpoints on 'rest_api_init'.
+	 *
+	 */
+	public function __construct() {
+		$this->namespace = Manager::ROOT_NAMESPACE . '/v' . Manager::VERSION;
+		$this->rest_base = Manager::REST_BASE . $this->get_name();
```

Comment:
Make it extandable

Reply:
Handled https://github.com/elementor/elementor/pull/14205/commits/0bd0b5556a56f1f98cb1b4dffbfba5568d22d57d